integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextmove/v2/NextMoveMessageService.java

```diff
@@ -0,0 +1,106 @@
+import no.difi.meldingsutveksling.noarkexchange.receive.InternalQueue;
+import no.difi.meldingsutveksling.receipt.ConversationService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+@Component
+@Slf4j
+@RequiredArgsConstructor
+public class NextMoveMessageService {
+
+    private final NextMoveValidator validator;
+    private final NextMoveOutMessageFactory nextMoveOutMessageFactory;
+    private final CryptoMessagePersister cryptoMessagePersister;
+    private final NextMoveMessageOutRepository messageRepo;
+    private final InternalQueue internalQueue;
+    private final ConversationService conversationService;
+
+    NextMoveOutMessage getMessage(String messageId) {
+        return messageRepo.findByMessageId(messageId)
+                .orElseThrow(() -> new MessageNotFoundException(messageId));
+    }
+
+    Page<NextMoveOutMessage> findMessages(Predicate predicate, Pageable pageable) {
+        return messageRepo.findAll(predicate, pageable);
+    }
+
+    public NextMoveOutMessage createMessage(StandardBusinessDocument sbd, List<? extends MultipartFile> files) {
+        NextMoveOutMessage message = createMessage(sbd);
+        files.forEach(file -> addFile(message, file));
+        return message;
+    }
+
+    public NextMoveOutMessage createMessage(StandardBusinessDocument sbd) {
+        validator.validate(sbd);
+        NextMoveOutMessage message = nextMoveOutMessageFactory.getNextMoveOutMessage(sbd);
+        messageRepo.save(message);
+        conversationService.registerConversation(message);
+        return message;
+    }
+
+    public void addFile(NextMoveOutMessage message, MultipartFile file) {
+        validator.validateFile(message, file);
+
+        String identifier = persistFile(message, file);
+
+        message.addFile(new BusinessMessageFile()
+                .setIdentifier(identifier)
+                .setTitle(getTitle(file.getName()))
+                .setFilename(file.getOriginalFilename())
+                .setMimetype(getMimeType(file.getContentType(), file.getOriginalFilename()))
+                .setPrimaryDocument(message.isPrimaryDocument(file.getOriginalFilename())));
+
+        messageRepo.save(message);
+    }
+
+    private String getTitle(String name) {
+        return StringUtils.hasText(name) ? name : null;
+    }
+
+    private String persistFile(NextMoveOutMessage message, MultipartFile file) {
+        String identifier = UUID.randomUUID().toString();
+
+        try {
+            cryptoMessagePersister.writeStream(message.getMessageId(), identifier, file.getInputStream(), file.getSize());
+        } catch (IOException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1166)